### ISSUE TITLE
Logrotate-related crash with icinga2 2.11.3-1

### ISSUE THREAD

**Issue by K0nne** [Icinga/icinga2#8050]
## Describe the bug

Last night one of our icinga2 satellites died. Application restart and logrotate for icinga2.log happened in the same minute. I found the following log entries:

**messages**
```
Jun  9 03:15:05 satellite02 logrotate: ALERT exited abnormally with [1]
Jun  9 03:15:07 satellite02 rhsmd: In order for Subscription Manager to provide your system with updates, your system must be registered with the Customer Portal. Please enter your Red Hat login to ensure your system is up-to-date.
Jun  9 03:15:49 satellite02 systemd: icinga2.service: main process exited, code=exited, status=138/n/a
Jun  9 03:15:49 satellite02 systemd: Unit icinga2.service entered failed state.
Jun  9 03:15:49 satellite02 systemd: icinga2.service failed.
```

**icinga2.log**

```
[2020-06-09 03:14:44 +0200] information/Application: Received request to shut down.
[2020-06-09 03:14:44 +0200] information/Application: Shutting down...
[2020-06-09 03:15:39 +0200] warning/JsonRpcConnection: API client disconnected for identity 'master02'
[2020-06-09 03:15:39 +0200] warning/ApiListener: Removing API client for endpoint 'master02'. 0 API clients left.
[2020-06-09 03:15:41 +0200] information/CheckerComponent: 'checker' stopped.
[2020-06-09 03:15:41 +0200] information/ApiListener: 'api' stopped.

```

## To Reproduce

It seems logrotete-related to me.

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`):

**satellite**
```
icinga2 - The Icinga 2 network monitoring daemon (version: 2.11.3-1)

Copyright (c) 2012-2020 Icinga GmbH (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: Red Hat Enterprise Linux Server
  Platform version: 7.8 (Maipo)
  Kernel: Linux
  Kernel version: 3.10.0-1127.el7.x86_64
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.5
  Build host: runner-LTrJQZ9N-project-322-concurrent-0

Application information:

General paths:
  Config directory: /etc/icinga2
  Data directory: /var/lib/icinga2
  Log directory: /var/log/icinga2
  Cache directory: /var/cache/icinga2
  Spool directory: /var/spool/icinga2
  Run directory: /run/icinga2

Old paths (deprecated):
  Installation root: /usr
  Sysconf directory: /etc
  Run directory (base): /run
  Local state directory: /var

Internal paths:
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid
```

* Enabled features (`icinga2 feature list`):

**satellite**
```
Disabled features: command compatlog debuglog elasticsearch gelf graphite influxdb livestatus notification opentsdb perfdata statusdata syslog
Enabled features: api checker mainlog
```

* Icinga Web 2 version and modules (System - About):

```
Icinga Web 2 Version
    2.7.3
Git Commit
    06cabfe8ba28cf545a42c92f25484383191a4e51
PHP Version
    7.1.8
Git Commit Datum
    2019-10-18
Copyright
    © 2013-2020 Das Icinga Projekt 
```

```
businessprocess 	2.2.0
graphite 	1.1.0
ipl 	v0.3.0
monitoring 	2.7.3
own module 	1.6 
```

* Config validation (`icinga2 daemon -C`):

**satellite**

```
[2020-06-09 09:43:14 +0200] information/cli: Icinga application loader (version: 2.11.3-1)
[2020-06-09 09:43:14 +0200] information/cli: Loading configuration file(s).
[2020-06-09 09:43:16 +0200] information/ConfigItem: Committing config item(s).
[2020-06-09 09:43:16 +0200] information/ApiListener: My API identity: satellite02
[2020-06-09 09:43:16 +0200] warning/CheckerComponent: Attribute 'concurrent_checks' for object 'checker' of type 'CheckerComponent' is deprecated and should not be used.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 76 HostGroups.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 1 FileLogger.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 1 IcingaApplication.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 12497 Hosts.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 1 ApiListener.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 2 Comments.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 1 CheckerComponent.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 14 Zones.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 12 Endpoints.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 305 CheckCommands.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 6 TimePeriods.
[2020-06-09 09:43:21 +0200] information/ConfigItem: Instantiated 48246 Services.
[2020-06-09 09:43:21 +0200] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2020-06-09 09:43:21 +0200] information/cli: Finished validating the configuration file(s).
```

**Comment by lippserd** [comment]
Hi @K0nne,

Any chance that you could provide a core dump here?

All the best,
Eric

**Comment by K0nne** [comment]
Hi @lippserd ,

I will try to reproduce it in our dev-environment.

**Comment by K0nne** [comment]
Unfortunately I am unable to reproduce the crash.

**Comment by airbone95** [comment]
Hi,

i also have this Problem on `r2.11.2-1` and also on `r2.12.0-rc1-1`. It looks like this happens if Icinga2 tries to reload (in my case due to new Config) and Logrotate runs at the same time.

I try to provide more information if i'm able to gather more.

**Comment by airbone95** [comment]
Hi,

i was able to reproduce this. This happens if Icinga2 receives a SIGHUP (for reloading) and a SIGUSR1 (for logrotate) at the same time.

To reproduce this i did this:
```bash
# kill -HUP $(cat /run/icinga2/icinga2.pid); kill -USR1 $(cat /run/icinga2/icinga2.pid)
```
and a few seconds later (about the time the configcheck usually takes) Icinga2 dies with Exitcode 138:
```bash
# journalctl -u icinga2.service
Jul 02 08:43:25 master02vt systemd[1]: icinga2.service: Main process exited, code=exited, status=138/n/a
Jul 02 08:43:25 master02vt systemd[1]: icinga2.service: Killing process 44817 (icinga2) with signal SIGKILL.
Jul 02 08:43:25 master02vt systemd[1]: icinga2.service: Failed with result 'exit-code'.
```

`icinga2 -V`:
```
icinga2 - The Icinga 2 network monitoring daemon (version: r2.12.0-rc1-1)

Copyright (c) 2012-2020 Icinga GmbH (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: Debian GNU/Linux
  Platform version: 10 (buster)
  Kernel: Linux
  Kernel version: 4.19.0-9-amd64
  Architecture: x86_64

Build information:
  Compiler: GNU 8.3.0
  Build host: runner-LTrJQZ9N-project-298-concurrent-0
  OpenSSL version: OpenSSL 1.1.1d  10 Sep 2019

Application information:

General paths:
  Config directory: /etc/icinga2
  Data directory: /var/lib/icinga2
  Log directory: /var/log/icinga2
  Cache directory: /var/cache/icinga2
  Spool directory: /var/spool/icinga2
  Run directory: /run/icinga2

Old paths (deprecated):
  Installation root: /usr
  Sysconf directory: /etc
  Run directory (base): /run
  Local state directory: /var

Internal paths:
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid
```

This is my Logrotate-Config:
```bash
# mainlog
/var/log/icinga2/icinga2.log {
  daily
  rotate 365
  compress
  create 0644 nagios nagios
  postrotate
    /bin/kill -USR1 $(cat /run/icinga2/icinga2.pid 2> /dev/null) 2> /dev/null || true
  endscript
}
```

**Comment by Al2Klimov** [comment]
Even 844e821cd is affected.

**Comment by Al2Klimov** [comment]
@N-o-X Shall I backport #8166? To which version?

**Comment by N-o-X** [comment]
Closed via #8227 (GitHub didn't close it)
